tests/system/model_monitoring/test_model_monitoring.py

```diff
@@ -1432,19 +1432,18 @@ def test_inference_feature_set(self) -> None:
         #     **({} if self.image is None else {"image": self.image}),
         # )
 
-        mlrun.model_monitoring.api.record_results(
+        model_endpoint = mlrun.model_monitoring.api.record_results(
             project=self.project_name,
             model_path=self.project.get_artifact_uri(
                 key=self.model_name, category="model", tag="latest"
             ),
             model_endpoint_name=self.model_endpoint_name,
-            function_name=self.function_name,
             context=mlrun.get_or_create_ctx(name=f"{self.name_prefix}-context"),  # pyright: ignore[reportGeneralTypeIssues]
             infer_results_df=self.infer_results_df,
             # TODO: activate ad-hoc mode when ML-5792 is done
         )
 
-        self._test_feature_names()
+        self._test_feature_names(endpoint_id=model_endpoint.metadata.uid)
```

Comment:
if you implement my above suggestion then it would be
```suggestion
        self._test_feature_names(endpoint_id=model_endpoint.metadata.uid, monitoring_feature_set_uri=model_endpoint.spec.monitoring_feature_set_uri)
```